File: creacion/factory_method/java/patterns_3_factory_method_solucion/src/test/java/org/formacion/factorymethod/TestLavadora.java
package org.formacion.factorymethod;

import static org.junit.Assert.*;

import org.junit.Test;

public class TestLavadora {

	@Test
	public void test_carga_frontal() {
		
		// Los dos metodos de test solo se diferencian en el tipo de factory usado.
		
		LavadoraFactory factory = new LavadoraCargaFrontalFactory();
		
		Lavadora lavadora = factory.crea();
		
		assertEquals("frontal", lavadora.tipoCarga);
		assertTrue(lavadora.tieneMandos);
		assertTrue(lavadora.tieneTambor);
	}

	@Test
	public void test_carga_superior() {
		
		LavadoraFactory factory = new LavadoraCargaSuperiorFactory();
		
		Lavadora lavadora = factory.crea();
		
		assertEquals("superior", lavadora.tipoCarga);
		assertTrue(lavadora.tieneMandos);
		assertTrue(lavadora.tieneTambor);
	}

}

File: creacion/factory_method/python/solution/src/test/testLavadora.py
#!/usr/bin/python3
# coding: utf-8

from unittest import TestCase
from ..main.classes.lavadoraCargaFrontalFactory import LavadoraCargaFrontalFactory
from ..main.classes.lavadoraCargaSuperiorFactory import LavadoraCargaSuperiorFactory

class TestLavadora(TestCase):

    def test_carga_frontal(self):
        factory = LavadoraCargaFrontalFactory()

        lavadora = factory.crea()

        self.assertEqual("frontal", lavadora.tipoCarga)
        self.assertTrue(lavadora.tieneMandos)
        self.assertTrue(lavadora.tieneTambor)

    def test_carga_superior(self):
        factory = LavadoraCargaSuperiorFactory()
        lavadora = factory.crea()

        self.assertEqual("superior", lavadora.tipoCarga)
        self.assertTrue(lavadora.tieneMandos)
        self.assertTrue(lavadora.tieneTambor)